docs/dataset_integration_guide.md

```diff
@@ -0,0 +1,120 @@
+* **Dataset** - a specific provider-model-variant. e.g. noaa-gfs-forecast
+
+
+
+## Integration steps
+
+### 1. Set up template
+
+#### Copy template
+From a terminal in the root of this repository, run these commands to copy a template of dataset integration code and tests.
+```bash
+DATASET_ID="<provider>-<model>-<variant>"
+DATASET_PATH="<provider>/<model>/<variant>"
+mkdir -p src/reformatters/$DATASET_PATH
+cp -r src/reformatters/example/ src/reformatters/$DATASET_PATH
+mkdir -p tests/$DATASET_PATH
+cp -r tests/example tests/$DATASET_PATH
+```
+
+#### Rename
+Run the following find/replaces. Follow PEP 8 on abbreviation capitalization in class names, e.g. `NoaaGfsForecastDataset`.
+
+Find and replace within `src/reformatters/$DATASET_PATH` and `tests/$DATASET_PATH`:
+1. `ExampleDataset` -> `<Producer><Model><Variant>Dataset`
+1. `ExampleTemplateConfig` -> `<Producer><Model><Variant>TemplateConfig`
+1. `ExampleRegionJob` -> `<Producer><Model><Variant>RegionJob`
+1. `ExampleDataVar` -> `<Producer>[<model>]DataVar`
+1. `ExampleInternalAttrs` -> `<Producer>[<model>]DataVar`
+1. `ExampleSourceFileCoord` -> `<Producer><Model>SourceFileCoord`
+1. `reformatters.example` -> `reformatters.<producer>.<model>.<variant>` (imports in tests)
+
+DataVar, InternalAttrs, and SourceFileCoord definitions can often be shared among mutiple datasets from the same producer.
+
+### 2. Register your dataset
+
+Add an instance of your `DynamicalDataset` subclass to the `DYNAMICAL_DATASETS` constant in `src/reformatters/__main__.py`:
+```python
+from reformatters.provider.model.variant import ProviderModelVariantDataset
+
+DYNAMICAL_DATASETS = [
+    ...,
+    ProviderModelVariantDataset(),
+]
+```
+
+### 3. Implement `TemplateConfig` subclass
+
+Work through `src/reformatters/$DATASET_PATH/template_config.py`, setting the attributes and method definitions to describe the structure of your dataset.
+
+Hint: providing an AI/LLM with 1) the example template config code to edit, 2) output of running `gdal-info <example source data file>` and 3) any dataset documentation will help it give you a decent first implementation of your `TemplateConfig` subclass.
+
+Using the information in the TemplateConfig, `reformatters` writes the Zarr metadata for your dataset to `src/reformatters/$DATASET_PATH/templates/latest.zarr`.  Run this command in your terminal to create or update the template based on the your `TemplateConfig` subclass:
+```bash
+uv run main $DATASET_ID update-template
+git add src/reformatters/$DATASET_PATH/templates/latest.zarr
+```
+Tracking the template in git lets us review diffs of any changes to the structure of our dataset.
+
+Run the tests, making any changes necessary.
+```bash
+uv run pytest tests/$DATASET_PATH/template_config_test.py
+```
+
+### 4. Implement `RegionJob` subclass
```

Comment:
Might be worth some words here or up in `Words` describing what exactly a `RegionJob` is. The base class doesn't have a docstring either so, I'm not sure it's spelled out anywhere (though probably can be inferred)